src/ocean/meta/traits/Function.d

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+
+    Traits specific to various function-like types
+
+    Copyright:
+        Copyright (c) 2017 sociomantic Labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.meta.traits.Function;
+
+import ocean.meta.AliasSeq;
+import ocean.meta.traits.Basic;
+
+/*******************************************************************************
+
+    Evaluates to a compile-time sequence representing the parameters of Fn
+
+    Params:
+        Fn = callable type (function/delegate/function pointer)
+
+*******************************************************************************/
+
+template ParametersOf ( Fn )
+{
+    static if (is( Fn Params == function ))
+        alias AliasSeq!(Params) ParametersOf;
+    else static if (is( Fn Params == delegate ))
+        alias AliasSeq!(ParametersOf!(Params)) ParametersOf;
+    else static if (is( Fn Params == Params* ))
```

Comment:
Suggest replacing `Params` with a name that describes what it really is:
- `is( Fn Params == delegate )`: not `Params` but `DgFuncptr`
- `is( Fn Params == Params* )`: not `Params` but `Function`